qt/scientific_interfaces/Inelastic/Processor/ElwinView.cpp

```diff
@@ -205,6 +205,14 @@ void ElwinView::addTableEntry(int row, std::string const &name, std::string cons
   setCell(std::move(cell), row, 1);
 }
 
+void ElwinView::updatePreviewWorkspaceNames(const std::vector<std::string> &names) {
+  disconnectSignals();
```

Comment:
Uhm I noticed that if you remove `disconnectSignals` and `connectSignals` here, then the preview box triggers the signal `currentIndexChanged(int)` which in turn calls the slot `notifyPreviewIndexChanged` which calls the handler `handlePreviewIndexChanged` which will call the function that sets the input workspace `setInputWorkspace(workspace)` and other very useful stuff like plot inputs. This means that then you don't need to call `setInputWorkspace(workspace)` inside `newInputDataFromDialog()`. 

So you can have the function `newInputDataFromDialog()` call only `m_view->updatePreviewWorkspaceNames` and from then onward everything is handled by a chain of signals and slots

I have tried this and everything seems to work fine when a new workspace is added. I am only suggesting because I think it would be a bit cleaner this way.

Reply:
Thanks for picking this up.